clients/src/main/java/org/apache/kafka/common/requests/ApiVersionsResponse.java

```diff
@@ -220,4 +217,16 @@ private static FinalizedFeatureKeyCollection createFinalizedFeatureKeys(
 
         return converted;
     }
+
+    public static Optional<ApiVersionsResponseData.ApiVersion> intersect(ApiVersionsResponseData.ApiVersion thisVersion,
+                                                                         ApiVersionsResponseData.ApiVersion other) {
+        if (other == null) return Optional.empty();
```

Comment:
It's a bit weird that we only check if `other` is null. Do we need this check at all?

Reply:
From the usage of this method, only ```other``` can be null. So current check is enough.

Reply:
That's not a good way to design apis, right? This kind of thing should be symmetric. Why is `other` sometimes `null`?

Reply:
> Why is other sometimes null?

The api key is supported by client-side but Controller doesn't.

> That's not a good way to design apis, right? This kind of thing should be symmetric. 

make sense. will address it and add unit tests.